packages/astro/src/core/routing/manifest/create.ts

```diff
@@ -443,6 +462,18 @@ function createRedirectRoutes(
 			);
 		}
 
+		validateRedirectPaths(from, destination);
+
+		const redirectRoute =
+			routeMap.get(destination) ??
+			[...routeMap.values()]
+				.sort(routeComparator)
+				.find(
+					(route) =>
+						route.pattern.test(destination) ||
+						route.fallbackRoutes.some((fallbackRoute) => fallbackRoute.pattern.test(destination))
```

Comment:
This is basically the same as matchRoute, but the current matchRoute code in match.js is for handling requests one level higher, and thus requires the entire Manifest